service/src/main/kotlin/fi/espoo/evaka/daycare/controllers/UnitAclController.kt

```diff
@@ -605,4 +620,46 @@ class UnitAclController(private val accessControl: AccessControl) {
 
     fun parseCoefficientValue(bool: Boolean) =
         if (bool) coefficientPositiveValue else coefficientNegativeValue
+
+    init {
+        asyncJobRunner.registerHandler(::deletePersonalMobileDevicesIfNeeded)
+    }
+
+    private fun deletePersonalMobileDevicesIfNeeded(
+        db: Database.Connection,
+        clock: EvakaClock,
+        job: AsyncJob.DeletePersonalDevicesIfNeeded,
+    ) {
+        db.transaction { tx ->
+            if (
+                !tx.hasRoleInAnyUnitWithProviderType(
+                    job.employeeId,
+                    UserRole.UNIT_SUPERVISOR,
+                    ProviderType.MUNICIPAL,
```

Comment:
should this also have `setOf(ProviderType.MUNICIPAL, ProviderType.MUNICIPAL_SCHOOL)`?

Reply:
Fixed 👍

Reply:
is it?

Reply:
Fixed 2 out of 3 places earlier. Now all places are fixed 👍